plugins/catalog-backend-module-gitlab/src/GitLabDiscoveryProcessor.ts

```diff
@@ -135,6 +145,8 @@ export class GitLabDiscoveryProcessor implements CatalogProcessor {
       );
     }
 
+    await this.cache.set(this.getCacheKey(), new Date().toISOString());
```

Comment:
Feels safer to grab this timestamp before making the request, so that we err on the side of requesting too much rather than too little.

That is:

```
lastActivity = cache.get(...)
now = new Date().toISOString()
// request
cache.set(..., lastActivity || now)
```

Reply:
I think there's a problem with this as the last activity will never change from its original value if we just set it to whatever existed in the cache.

Reply:
Yep it needs some adjustment, whatever works, but I'd like to avoid the race

Reply:
@Rugvip I'm thinking either we make it so that it respects the last modified or we simply don't cache it at all. If we want to give it some leeway adjusting the time we cache by offsetting it by an arbitrary amount of time. I'm not sure what we'd achieve by setting it to a virtually static value.

Reply:
Let's say we execute a request at the arbitrary point in time `15`. The response takes a little time to be processed and returned, so it arrives at `19`. That's the point when we call `Date.now()` in our code, so we store `19` in the cache. The next request is then going to ask for any updates that happened after `19`.

Now let's say there was activity at `18`, which we won't be included in the initial request because our request reached GitLab before that. We'll also miss it in our next request though, since we're asking for updates that happened after `19`.

The fix I'm after is to grab the timestamp before the request rather than after, so that what we store in the cache in the above example is `15`, rather than `19`, meaning there's no gap in time where we would miss updates.

Reply:
Ah, alright, I see! Can do.

Reply:
I've made a fix and rebased @Rugvip, please check if it's in line with what you expected.